test/unit/eval_algorithms/test_faithfulness.py

```diff
@@ -0,0 +1,413 @@
+        )
+        # THEN
+        assert test_case.expected_score == actual_score
+
+    @patch("fmeval.eval_algorithms.faithfulness.get_eval_results_path")
+    @patch("fmeval.eval_algorithms.faithfulness.evaluate_dataset")
+    @patch("fmeval.eval_algorithms.faithfulness.Faithfulness._build_pipeline")
+    @patch("fmeval.eval_algorithms.faithfulness.get_dataset")
+    @patch("fmeval.eval_algorithms.faithfulness.get_dataset_configs")
+    def test_evaluate(
+        self,
+        mock_get_dataset_configs,
+        mock_get_dataset,
+        mock_build_pipeline,
+        mock_evaluate_dataset,
+        mock_get_results_path,
+    ):
+        """
+        GIVEN a Faithfulness instance.
+        WHEN its evaluate method is called with valid arguments.
+        THEN `evaluate_dataset` is called with the correct arguments.
+        """
+        dataset_config = Mock()
+        dataset_config.dataset_name = "my_custom_dataset"
+        mock_get_dataset_configs.return_value = [dataset_config]
+
+        mock_dataset = Mock()
+        # So that validate_dataset does not error
+        mock_dataset.columns = Mock(
+            return_value=[
+                DatasetColumns.MODEL_INPUT.value.name,
+                DatasetColumns.MODEL_OUTPUT.value.name,
+                DatasetColumns.TARGET_CONTEXT.value.name,
+            ]
+        )
+        mock_get_dataset.return_value = mock_dataset
+
+        mock_get_results_path.return_value = "/path/to/results"
+        model_runner = Mock()
+
+        mock_build_pipeline.return_value = Mock()
+        # WHEN
+        faithfulness_algo = Faithfulness()
+        output = faithfulness_algo.evaluate(
+            judge_model=model_runner,
+            dataset_config=dataset_config,
+            num_records=162,
+            save=True,
+        )
+        # THEN
+        mock_build_pipeline.assert_called_with(
+            model_runner,
+            LONG_FORM_ANSWER_PROMPT,
+            NLI_STATEMENTS_MESSAGE,
+        )
+        mock_evaluate_dataset.assert_called_once_with(
+            dataset=mock_dataset,
+            pipeline=mock_build_pipeline.return_value,
+            dataset_name=dataset_config.dataset_name,
+            eval_name=faithfulness_algo.eval_name,
+            metric_names=["faithfulness"],
+            eval_results_path="/path/to/results",
+            agg_method=MEAN,
+            save=True,
+            save_strategy=None,
+        )
+        assert output == [mock_evaluate_dataset.return_value]
+
+    @patch("fmeval.eval_algorithms.faithfulness.get_eval_results_path")
+    @patch("fmeval.eval_algorithms.faithfulness.evaluate_dataset")
+    @patch("fmeval.eval_algorithms.faithfulness.Faithfulness._build_pipeline")
+    @patch("fmeval.eval_algorithms.faithfulness.get_dataset")
+    @patch("fmeval.eval_algorithms.faithfulness.get_dataset_configs")
+    @patch("fmeval.eval_algorithms.faithfulness.get_default_judge_model")
+    def test_evaluate_with_default_judge_model(
```

Comment:
Suggestion: can you parametrize the test above so that you don't need to have an entirely new test function just for the `get_default_judge_model` validation?